File: Dave Min/2023-05-11-Invert Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        TreeNode node = new TreeNode();
        node = root;
         doInvertTree(node);
         return root;
    }
    public void doInvertTree(TreeNode node) {
        if(node==null) return;

        TreeNode tmpL = node.left;
        TreeNode tmpR = node.right;
        node.left = tmpR;
        node.right = tmpL;
        
        doInvertTree(node.left);
        doInvertTree(node.right);
    }   
}

File: Hyunjin Kim/2023-05-11-invert-binary-tree.py
from typing import Optional
# Definition for a binary tree node.
from typing import Optional

class TreeNode:
     def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        #go to the right side first and print. if there isn't any,
        if not root :
            return None

        root.left, root.right = self.invertTree(root.right), self.invertTree(root.left)

        return root
